contracts/poolMaster/PoolMaster.sol

```diff
@@ -0,0 +1,329 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.7.6;
+
+import '@openzeppelin/contracts/math/SafeMath.sol';
+import {IERC20Ext} from '@kyber.network/utils-sc/contracts/IERC20Ext.sol';
+import {ERC20, ERC20Burnable} from '@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol';
+import {SafeERC20} from '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import {ReentrancyGuard} from '@openzeppelin/contracts/utils/ReentrancyGuard.sol';
+import {
+  PermissionAdmin,
+  PermissionOperators
+} from '@kyber.network/utils-sc/contracts/PermissionOperators.sol';
+
+import {IKyberStaking} from '../interfaces/staking/IKyberStaking.sol';
+import {IRewardsDistributor} from '../interfaces/rewardDistribution/IRewardsDistributor.sol';
+import {IKyberGovernance} from '../interfaces/governance/IKyberGovernance.sol';
+
+interface INewKNC {
+  function mintWithOldKnc(uint256 amount) external;
+
+  function oldKNC() external view returns (address);
+}
+
+interface IKyberNetworkProxy {
+  function swapEtherToToken(IERC20Ext token, uint256 minConversionRate)
+    external
+    payable
+    returns (uint256 destAmount);
+
+  function swapTokenToToken(
+    IERC20Ext src,
+    uint256 srcAmount,
+    IERC20Ext dest,
+    uint256 minConversionRate
+  ) external returns (uint256 destAmount);
+}
+
+contract PoolMaster is PermissionAdmin, PermissionOperators, ReentrancyGuard, ERC20Burnable {
+  using SafeMath for uint256;
+  using SafeERC20 for IERC20Ext;
+
+  struct Fees {
+    uint256 mintFeeBps;
+    uint256 claimFeeBps;
+    uint256 burnFeeBps;
+  }
+
+  enum FeeTypes {MINT, CLAIM, BURN}
+  IERC20Ext internal constant ETH_ADDRESS = IERC20Ext(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+  uint256 internal constant PRECISION = (10**18);
+  uint256 internal constant BPS = 10000;
+  uint256 internal constant INITIAL_SUPPLY_MULTIPLIER = 10;
+  Fees public adminFees;
+  uint256 public withdrawableAdminFees;
+
+  IKyberNetworkProxy public kyberProxy;
+  IKyberStaking public immutable kyberStaking;
+  IRewardsDistributor public rewardsDistributor;
+  IKyberGovernance public kyberGovernance;
+  IERC20Ext public newKnc;
+  IERC20Ext private oldKnc;
+
+  receive() external payable {}
+
+  constructor(
+    string memory _name,
+    string memory _symbol,
+    IKyberNetworkProxy _kyberProxy,
+    IKyberStaking _kyberStaking,
+    IKyberGovernance _kyberGovernance,
+    IRewardsDistributor _rewardsDistributor,
+    uint256 _mintFeeBps,
+    uint256 _claimFeeBps,
+    uint256 _burnFeeBps
+  ) ERC20(_name, _symbol) PermissionAdmin(msg.sender) {
+    kyberProxy = _kyberProxy;
+    kyberStaking = _kyberStaking;
+    kyberGovernance = _kyberGovernance;
+    rewardsDistributor = _rewardsDistributor;
+    newKnc = IERC20Ext(address(_kyberStaking.kncToken()));
```

Comment:
```
address _newKnc = IERC20Ext(address(_kyberStaking.kncToken()));
newKnc =_newKnc;
...
_newKnc.safeApprove(address(_kyberStaking), type(uint256).max);
```

this way you can make newKnc and oldKnc are immutable

Reply:
yes, they are unchangeable, so should make them immutable as well